packages/@sanity/base/src/components/previews/general/MediaPreview.tsx

```diff
@@ -0,0 +1,92 @@
+import React, {useMemo} from 'react'
+import {getDevicePixelRatio} from 'use-device-pixel-ratio'
+import {Skeleton, Stack, Text, Tooltip} from '@sanity/ui'
+import {CircularProgress} from '../../progress'
+import {Media} from '../_common/Media'
+import {PREVIEW_MEDIA_SIZE} from '../constants'
+import {PreviewMediaDimensions, PreviewProps} from '../types'
+import {MediaWrapper, Root, ProgressWrapper} from './MediaPreview.styled'
+
+interface MediaPreviewProps extends PreviewProps<'media'> {
+  withRadius?: boolean
+  withBorder?: boolean
+}
+
+const DEFAULT_MEDIA_DIMENSIONS: PreviewMediaDimensions = {
+  ...PREVIEW_MEDIA_SIZE.media,
+  aspect: 1,
+  fit: 'crop',
+  dpr: getDevicePixelRatio(),
+}
+
+export function MediaPreview(props: MediaPreviewProps) {
+  const {
+    media,
+    mediaDimensions = DEFAULT_MEDIA_DIMENSIONS,
+    children,
+    isPlaceholder,
+    progress = -1,
+    subtitle,
+    title,
+    withBorder = true,
+    withRadius = true,
+    ...restProps
+  } = props
+
+  const aspect = mediaDimensions.aspect
+
+  const STYLES_PADDER = useMemo(() => ({paddingBottom: `${100 / aspect}%`}), [aspect])
+
+  const tooltipContent = useMemo(() => {
+    if (!title || !subtitle) {
+      return null
+    }
+
+    return (
+      <Stack padding={2} space={2}>
+        {title && (
+          <Text align="center" size={1} weight="semibold">
+            {typeof title === 'function' ? title({layout: 'media'}) : title}
+          </Text>
+        )}
+
+        {subtitle && (
+          <Text align="center" muted size={1}>
+            {typeof subtitle === 'function' ? subtitle({layout: 'media'}) : subtitle}
+          </Text>
+        )}
+      </Stack>
+    )
+  }, [subtitle, title])
+
+  return (
+    <Root overflow="hidden" flex={1} {...restProps}>
+      <div style={STYLES_PADDER} />
+
+      <Tooltip content={tooltipContent} disabled={!tooltipContent} placement="top" portal>
+        <MediaWrapper align="center" justify="center">
+          {isPlaceholder ? (
+            <Skeleton animated radius={2} style={{width: '100%', height: '100%'}} />
```

Comment:
This can be rewritten into a styled component to get rid of inline styling.

```js
const MediaSkeleton = styled(Skeleton)`
  width: 100%;
  height: 100%;
`
```

Maybe we should update `Skeleton` with a `width` and `height` prop, since you always need to set a size of the component when you use it.

```js
<Skeleton animated radius={2} height="100%" width="100%" />
```